### PR TITLE
data: Refactor video frame and pixels representations

### PR DESCRIPTION
**Pull request by Luni-4**
This PR:

- Adds colorspace data and methods
- Refactors pixels structures

### CODE REVIEW

File: data/src/format.rs [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+use crate::plane::PlaneId;
+
+/// Video frame color space representation such as color models, primaries,
+/// color space conversion functions, and chroma location.
+///
+/// This representation also contains additional information about pixels planes
+/// such as their chroma definition.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct VideoFrameFormat {
```
Comment (lu-zero) [inline data/src/format.rs:20]: This is a Pixel format, the name is wrong.

File: data/src/frame.rs [hunk 2]
```diff
@@ -60,6 +60,43 @@ impl fmt::Display for FrameType {
     }
 }
 
+/// Plane identifier.
+#[derive(Clone, Copy)]
+pub enum PlaneId {
```
Comment (lu-zero) [inline data/src/frame.rs:65]: This abstraction is unnecessary, please do not introduce it.

**Review by lu-zero — CHANGES_REQUESTED** [review]
The commit is too big and conflates too many not really related changes.

In general some renames might be good, some aren't.

The additional abstractions you introduced should be avoided.

File: data/src/format.rs (hunk 1 above)
Comment (Luni-4) [reply data/src/format.rs:20]: Just a small explanation on the reasons of this movement. 
 -`pixel.rs` contains color spaces structures and functions
- `format.rs` contains frame pixels representations. My idea would be to add `Soniton`, renamed ad `AudioSampleFormat`, at a second time.

For what concerns the name of the structure, can we use `PixelFrameFormat`?

Comment (lu-zero) [reply data/src/format.rs:20]: The current names might be a bit unique but they are short and to the point. Renaming them with with something that's longer and/or misleading isn't great.

`AudioFormat` or `format::Audio` and `PixelFormat` or `format::Pixel`, might be short enough, but I would not spend time now on renaming something that's already working.

**Comment by FreezyLemon** [comment]
Please reorganize the changes into multiple commits, it is difficult to review the PR like this. It might even be too much for one PR, but we can see when it's more organized

File: data/src/format.rs (hunk 1 above)
Comment (Luni-4) [reply data/src/format.rs:20]: I think I will adopt your `format::name` idea, thanks!

**Comment by Luni-4** [comment]
Ok, I should have implemented the requests and splitted in more commits

File: data/src/pixel.rs [hunk 3]
```diff
@@ -1,691 +1,292 @@
...
+    }
+
+    /// Pixel data is paletted.
+    pub const fn is_paletted(mut self) -> Self {
+        self.is_paletted = true;
+        self
+    }
+
+    /// Adds first chromaton pixel plane.
+    pub const fn add_first_plane(mut self, chroma_data: Chromaton) -> Self {
```
Comment (lu-zero) [inline data/src/pixel.rs:246]: This is still wrong.

Comment (Luni-4) [reply data/src/pixel.rs:246]: Do you have an alternative to propose? Can you explain why it is wrong?

File: data/src/pixel.rs [hunk 4]
```diff
@@ -1,691 +1,292 @@
...
+/// color space conversion functions, and chroma location.
 ///
-/// For example, the format can be paletted, so the components describe
-/// the palette storage format, while the actual data is 8-bit palette indices.
+/// This representation also contains additional information about pixels
+/// such as their chroma definition.
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
-pub struct Formaton {
-    /// Image color model.
+pub struct Pixel {
```
Comment (lu-zero) [inline data/src/pixel.rs:157]: I said already once that this should be postponed.

File: data/src/colorspace.rs [hunk 5]
```diff
@@ -0,0 +1,448 @@
+//! Expose all data and methods to interact with color spaces.
```
Comment (lu-zero) [inline data/src/colorspace.rs:1]: If you are moving out code, you should make so it is evident in the commit.

**Review by lu-zero — CHANGES_REQUESTED** [review]
Please do not rename the Formaton, nor all those changes in the related commit.

**Pull request closed without merging**